src/visx-visualization/controls/Controls.tsx

```diff
@@ -503,6 +505,16 @@ export function ViewTypeControl() {
   const restorePreviousTopGraphType = useRestorePreviousTopGraphType();
   const trackEvent = useTrackEvent();
 
+  const transposeData = useTranspose();
+  const swapAxisConfigs = useSwapAxisConfigs();
+
+  const hasBeenTransposed = useHasBeenTransposed();
+
+  const handleTranspose = useEventCallback(() => {
+    transposeData();
+    swapAxisConfigs();
+  });
```

Comment:
I see what you mean about duplication of logic between the context menu and controls. I think defining this as a separate hook in a new file in the `hooks` directory would make sense (rather than an additional context) since its dependencies are already in a context and this functionality is entirely derived from the upper states.

So this can be pulled out into a `useHandleTranspose.ts` file that would look like this:


```
export function useHandleTranspose() {
	const yScale = useYScale();
  const expandedValues = useSelectedValues();
	const transposeData = useTranspose();
	const swapAxisConfigs = useSwapAxisConfigs();
 		
	return useEventCallback(() => {
		// Perform data transposition operations
		transposeData();
		swapAxisConfigs();
		// Reset scroll state to avoid invalid viewport
		xScale.resetScroll();
    yScale.resetScroll();
    // Reset expanded rows since they are no longer relevant
    expandedValues.reset();
	});
}
```

Which would then be consumed here in the Controls and in the ContextMenu as simply


```suggestion
	// TODO: import the hook at the top of the file as well
	const handleTranspose = useHandleTranspose();
```